recipes/qpdf/all/conanfile.py

```diff
@@ -0,0 +1,168 @@
+            self.requires("openssl/1.1.1q")
+        elif self.options.with_crypto == "gnutls":
+            raise ConanInvalidConfiguration("GnuTLS is not available in Conan Center yet.")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.4")
+        elif self.options.with_jpeg == "mozjpeg":
+            self.requires("mozjpeg/4.0.0")
+
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        if is_msvc(self):
+            check_min_vs(self, "150")
+        else:
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.24.1")
+        self.tool_requires("pkgconf/1.9.3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_STATIC_LIBS"] = not self.options.shared
+        # https://qpdf.readthedocs.io/en/latest/installation.html#build-time-crypto-selection
+        tc.variables["USE_IMPLICIT_CRYPTO"] = False
+        tc.cache_variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        if self.options.with_crypto == "native":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = True
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = False
+        elif self.options.with_crypto == "openssl":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = False
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = True
+        tc.generate()
+        # TODO: after https://github.com/conan-io/conan/issues/11962 is solved
+        # we might obsolete here cmake deps generation and cmake patching and get
+        # the possibility to go for to pkg_config based dependency discovery instead.
+        # At the moment, even with the linked work-around, the linkage is mixed-up
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    def _patch_sources(self):
+        # patch 0001 and 0002 are uniquely appliable, based ony dependency config
+        patches_to_apply = ["0003-exclude-unnecessary-cmake-subprojects.patch"]
+        if self.options.with_crypto == "openssl":
+            patches_to_apply.append("0002-libqpdf-cmake-deps-jpeg-zlib-openssl.patch")
```

Comment:
Would it be better to always apply the other patch, and than conditionally add in openssl?

```py
apply_conandata_patches(self)
if self.options.with_crypto == "openssl":
    replace_in_files(self, source_folder + "cmakelists", 
                                "PUBLIC JPEG::JPEG ZLIB::ZLIB", "PUBLIC JPEG::JPEG ZLIB::ZLIB OpenSSL::SSL")
```

🤔 not sure it's better but just tossing ideas

Reply:
Yeah i think your code here is not complete, since the found_package needs to be patched also, but i really like the idea. This would avoid the code-duplication the patch-files. I will update it accordingly